### ISSUE TITLE
yesod-test: byLabelExact fails to find element when using aria-label

### ISSUE THREAD

**Issue by chekkan** [yesodweb/yesod#1678]
`byLabelExact` fails to find element if the `aria-label` attribute is used. One of the approaches recommended in [w3.org](https://www.w3.org/WAI/tutorials/forms/labels/#hiding-label-text)

```html
<input type="email" name="email" placeholder="Email Address" aria-label="Email">
```

```hs
it "some test" $ do
      get HomeR
      request $ do
        setMethod "POST"
        setUrl SubscribeR
        byLabelExact "Email" "test@test.com"
```

```
Failures:

  ./Yesod/Test.hs:1568:28:
  1) Handler.Home.Homepage can subscribe with email
       No label contained: Email
```

# Dependencies and versions
```
/usr/local/bin/stack
Version 2.3.1, Git revision de2a7b694f07de7e6cf17f8c92338c16286b2878 (8103 commits) x86_64 hpack-0.33.0
/usr/local/bin/stack
The Glorious Glasgow Haskell Compilation System, version 8.8.3
wai 3.2.2.1
yesod 1.6.0.1
yesod-core 1.6.18
yesod-form 1.6.7
yesod-bin version: 1.6.0.4
```

**Comment by snoyberg** [comment]
Current behavior seems correct to me, aria-label is a different attribute

**Comment by chekkan** [comment]
@snoyberg is there an alternative which can be used to achieve this then?

**Comment by snoyberg** [comment]
There are general purpose HTML analysis functions in the library. If you look at the implementation of `byLabelExact`, you can see how it is implemented and how it looks for the `label` attribute. Right now, there isn't a high level `byAriaLabelExact` function.

**Comment by jezen** [comment]
The `byLabelExact` function is indeed working correctly, so I'll close this.
